components/team_card/team_card_storage.lua

```diff
@@ -0,0 +1,100 @@
+---
+-- @Liquipedia
+-- wiki=commons
+-- page=Module:TeamCard/Storage
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+local Custom = require('Module:TeamCard/Custom')
+local String = require('Module:StringUtils')
+-- TODO: Once the Template calls are not needed (when RL has been moved to Module), deprecate Qualifier Module
+local Qualifier = require('Module:TeamCard/Qualifier')
+local Variables = require('Module:Variables')
+
+local TeamCardStorage = {}
+
+function TeamCardStorage.saveToLpdb(args, teamObject, players, playerPrize)
+	local team, teamTemplateName
+
+	if type(teamObject) == 'table' then
+		if teamObject.team2 or teamObject.team3 then
+			team = 'TBD'
+		else
+			teamTemplateName = teamObject.teamtemplate
+			team = teamObject.lpdb
+		end
+	end
+
+	local smwPrefix = args.smw_prefix or Variables.varDefault('smw_prefix') or ''
+
+	-- Setup LPDB Data
+	local lpdbData = {}
+	lpdbData = TeamCardStorage._addStandardLpdbFields(lpdbData, team, args, smwPrefix)
+	lpdbData.participanttemplate = teamTemplateName
+	lpdbData.players = players
+	lpdbData.individualprizemoney = playerPrize
+
+	-- If a custom override for LPDB exists, use it
+	lpdbData = Custom.adjustLpdb and Custom.adjustLpdb(lpdbData, team, args, smwPrefix) or lpdbData
+
+	-- Jsonify the json fields
+	lpdbData.extradata = mw.ext.LiquipediaDB.lpdb_create_json(lpdbData.extradata)
+	lpdbData.players = mw.ext.LiquipediaDB.lpdb_create_json(lpdbData.players)
+
+	-- Name must match prize pool insertion
+	local storageName = Custom.getLpdbObjectName and Custom.getLpdbObjectName(team, smwPrefix)
+						or TeamCardStorage._getLpdbObjectName(team, smwPrefix)
+
+	mw.ext.LiquipediaDB.lpdb_placement(storageName, lpdbData)
+end
+
+-- Adds basic lpdb fields
+function TeamCardStorage._addStandardLpdbFields(lpdbData, team, args, smwPrefix)
+	local title = mw.title.getCurrentTitle().text
+	local tournamentName = Variables.varDefault('tournament name pp') or Variables.varDefault('tournament_name')
+	local date = Variables.varDefault('tournament_date')
+	local startDate = Variables.varDefault('tournament_startdate', Variables.varDefault('tournament_sdate', date))
+	local endDate = Variables.varDefault('tournament_enddate', Variables.varDefault('tournament_edate', date))
+
+	lpdbData.participant = team
+	lpdbData.tournament = tournamentName or title
+	lpdbData.series = Variables.varDefault('tournament_series')
+	lpdbData.parent = Variables.varDefault('tournament_parent')
+	lpdbData.startdate = startDate
+	lpdbData.date = args.date or Variables.varDefault('enddate_' .. team .. smwPrefix .. '_date') or endDate
+	lpdbData.qualifier, lpdbData.qualifierpage, lpdbData.qualifierurl = Qualifier.parseQualifier(args.qualifier)
+
+	if team ~= 'TBD' then
```

Comment:
maybe uppercase the team entry before you compare it to `TBD`
(same in line 92)

Reply:
The calling Module has a load of dependencies that it needs to be upper case. I made a change in the calling module to ensure the tbd team is in upper case.